cmd/multiFaToChain/multiFaToChain.go

```diff
@@ -0,0 +1,187 @@
+					currBaseStats.QBases++
+					prevState = InTGap
+				} else {
+					log.Fatalf("Unrecognized state.")
+				}
+			case InTGap:
+				if currState == InAln { //if we go from a T gap to alignment, we write out the old baseStats and start a new block.
+					currAlignment = append(currAlignment, currBaseStats)
+					currBaseStats = chain.BaseStats{Size: 1, TBases: 0, QBases: 0} //size is 1 to count current position
+					prevState = InAln
+				} else if currState == InQGap { //I'm not positive this is allowed, but if we have a t gap that switches to a q gap we'll make it part of the same line.
+					prevState = InQGap
+					currBaseStats.TBases++
+				} else {
+					log.Fatalf("Unrecognized state.")
+				}
+			case InQGap:
+				if currState == InAln {
+					currAlignment = append(currAlignment, currBaseStats)
+					currBaseStats = chain.BaseStats{Size: 1, TBases: 0, QBases: 0}
+					prevState = InAln
+				} else if currState == InTGap { //again, not certain this can be allowed.
+					prevState = InTGap
+					currBaseStats.QBases++
+				}
+			default:
+				log.Fatalf("Unrecognized state.")
+			}
+		}
+	}
+	currAlignment = append(currAlignment, currBaseStats)
+
+	var TEnd = len(recordsCopy[0].Seq)
+	var QEnd = len(recordsCopy[1].Seq)
+	//if there are trailing Tbases or Qbases after the last alignment, we trim them away from the header
+	//both T and Q are obligately positive stranded, so no need to check for strand.
+	if currAlignment[len(currAlignment)-1].TBases > 0 {
+		TEnd -= currAlignment[len(currAlignment)-1].TBases
+	}
+
+	if currAlignment[len(currAlignment)-1].QBases > 0 {
+		QEnd -= currAlignment[len(currAlignment)-1].QBases
+	}
+
+	var currAnswer chain.Chain = chain.Chain{
+		Score:     100, //dummy value
+		TName:     tName,
+		TSize:     len(recordsCopy[0].Seq),
+		TStrand:   true,
+		TStart:    0,
+		TEnd:      TEnd,
+		QName:     qName,
+		QSize:     len(recordsCopy[1].Seq),
+		QStrand:   true,
+		QStart:    0,
+		QEnd:      QEnd,
+		Alignment: currAlignment,
+		Id:        1,
+	}
+
+	answer = append(answer, currAnswer)
+	chain.Write(outFile, answer, header)
+}
+
+func queryState(records []fasta.Fasta, index int) (State, bool) {
+	if dna.DefineBase(records[0].Seq[index]) || records[0].Seq[index] == dna.N || records[0].Seq[index] == dna.LowerN {
+		if dna.DefineBase(records[1].Seq[index]) || records[1].Seq[index] == dna.N || records[1].Seq[index] == dna.LowerN {
+			return InAln, false
+		} else if records[1].Seq[index] == dna.Gap {
+			return InQGap, false
+		} else {
+			log.Fatalf("Unrecognized dna base in the query sequence: %s.", dna.BaseToString(records[1].Seq[index]))
+		}
+	} else if records[0].Seq[index] == dna.Gap {
+		if dna.DefineBase(records[1].Seq[index]) || records[1].Seq[index] == dna.N || records[1].Seq[index] == dna.LowerN {
+			return InTGap, false
+		} else if records[1].Seq[index] == dna.Gap {
+			return InAln, true //inAln is a dummy here, we care about the bool return signaling a doubleGap in the alignment.
```

Comment:
Should double gap be another state?  It doesn't seem like you want two things returned, but rather you are expanding the number of states by adding a second return value.  Maybe we should actually expand the number of states?

Reply:
I had considered that initially, but we want to pass over a doubleGap without changing state (so a double gap in a tGap should remain a tGap after we pass the doubleGap, for example). This would require us to store two state variables. Since doubleGap is ignored in all scenarios, i think it makes sense to be a separate variable, distinct from the states.

Reply:
ok, I understand better now.  Sounds good.